src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -651,6 +651,14 @@ public Builder inheritProcessOutput(boolean inheritProcessOutput) {
     }
 
     /**
+     * Sets the desired Java SDK path, used in devappserver runs and App Engine standard staging.
+     */
+    public Builder javaSdkPath(Path javaSdkPath) {
```

Comment:
Hum, I always thought there was a Java System Development Kit, but turns out there's only a Java Development Kit. Thanks for the tip!

Your suggested solution sounds too engineered for the problem at hand, and it binds us too much to there being a separate process being ran by a specific JDK which, in an ideal world, would just take the form of a Java API call.

It would also require a serious overhaul of the current ProcessRunner architecture, which I'm not convinced is worth revising.

I've also had ideas regarding validators for appengine-plugins-core for a year now, that I was never able to get around. But I wouldn't want to bundle them with other entities, like runners.